src/output/srzip.c

```diff
@@ -632,6 +706,88 @@ static int zip_append_analog_queue(const struct sr_output *o,
 	return SR_OK;
 }
 
+// zip up our directory to a standard "sr" file.
+static gboolean zip_to_srfile(gchar *filename)
+{
+	gboolean ret = 1;
+
+	// "example.sr" --> "example.sr.dir"
+	gchar *dirname = g_strdup_printf("%s.dir", filename);
+	int res = g_rename(filename, dirname);
+	if (res) {
+		sr_err("Failed to rename %s to %s",
+		       filename,
+		       dirname);
+		ret = 0;
+		goto fail_dir_rename;
+	}
+
+	// "/foo/bar/example.sr" --> "../example.sr"
+	gchar *basename = g_path_get_basename(filename);
+	gchar *zipname = g_build_filename("..",
+	                                  basename,
+	                                  NULL);
+
+	const gchar *compression = g_getenv("SR_COMPRESSION_FLAG");
+	if (!compression) {
+		compression = "-2"; // fastest in my tests.
+	}
+
+	gchar *argv[] = { "/usr/bin/zip",
```

Comment:
I'm afraid, this is breaking windows support

Reply:
Should be fixed now.